test/tap/tap/Makefile

```
@@ -39,38 +42,42 @@ default: all
 all: libtap_mariadb.a libtap_mysql57.a libtap_mysql8.a \
-	libtap.a libtap.so libcpp_dotenv.so libre2.so
+	libtap.a libtap$(SHLIB_EXT) libcpp_dotenv$(SHLIB_EXT) libre2$(SHLIB_EXT)
 
-debug: OPT := $(STDCPP) -O0 -DDEBUG -ggdb -Wl,--no-as-needed $(WASAN)
-debug: libtap_mariadb.a libtap_mysql57.a libtap_mysql8.a libtap.a libtap.so
+DEBUG_OPT := $(STDCPP) -O0 -DDEBUG -ggdb $(WASAN)
+ifneq ($(UNAME_S),Darwin)
+DEBUG_OPT += -Wl,--no-as-needed
+endif
+debug: OPT := $(DEBUG_OPT)
+debug: libtap_mariadb.a libtap_mysql57.a libtap_mysql8.a libtap.a libtap$(SHLIB_EXT)
```

checkmake (0.3.2)

[warning] Target "debug" defined multiple times (lines 51 and 52).

(uniquetargets)